simulator-spring-boot/src/main/java/org/citrusframework/simulator/http/HttpOperationScenario.java

```diff
@@ -101,34 +91,52 @@ private Message receive(ScenarioRunner scenario,
             requestActionBuilder.message().dictionary(inboundDataDictionary);
         }
 
-        AtomicReference<Message> receivedMessage = new AtomicReference<>();
+        AtomicReference<HttpMessage> receivedMessage = new AtomicReference<>();
         requestActionBuilder.getMessageProcessors().add(
-            (message, context) -> receivedMessage.set(message));
+            (message, context) -> receivedMessage.set((HttpMessage)message));
 
         // Verify incoming request
-        scenario.$(requestActionBuilder);
+        scenarioRunner.$(requestActionBuilder);
 
         return receivedMessage.get();
     }
 
-    private void respond(ScenarioRunner scenario,
-        OpenApiServerActionBuilder openApiServerActionBuilder, Message receivedMessage) {
+    private void respond(ScenarioRunner scenarioRunner,
+        OpenApiServerActionBuilder openApiServerActionBuilder, HttpMessage receivedMessage) {
 
         HttpServerResponseActionBuilder responseBuilder = null;
         if (httpResponseActionBuilderProvider != null) {
-            responseBuilder = httpResponseActionBuilderProvider.provideHttpServerResponseActionBuilder(operation, receivedMessage);
+            responseBuilder = httpResponseActionBuilderProvider.provideHttpServerResponseActionBuilder(scenarioRunner, this, receivedMessage);
         }
 
-        HttpStatus httpStatus = response != null && response.getStatusCode() != null ? HttpStatus.valueOf(Integer.parseInt(response.getStatusCode())) : HttpStatus.OK;
-        responseBuilder = responseBuilder != null ? responseBuilder : openApiServerActionBuilder.send(
-            operation.operationId, httpStatus);
+        if (responseBuilder == null) {
+            responseBuilder = createRandomMessageResponseBuilder(openApiServerActionBuilder, receivedMessage.getAccept());
+        }
+
+        scenarioRunner.$(responseBuilder);
+    }
 
+    /**
+     * Creates a builder that creates a random message based on OpenApi specification.
+     * @param openApiServerActionBuilder
+     * @return
+     */
+    private HttpServerResponseActionBuilder createRandomMessageResponseBuilder(
+        OpenApiServerActionBuilder openApiServerActionBuilder, String accept) {
+        HttpServerResponseActionBuilder responseBuilder;
+
+        OasResponse response = determineResponse(accept);
+
+        HttpStatus httpStatus = response != null && response.getStatusCode() != null ? HttpStatus.valueOf(Integer.parseInt(response.getStatusCode())) : HttpStatus.OK;
```

Comment:
a named method for this condition would do good, I think. it's more understandable this way.